base/threading/scoped_blocking_call_internal.cc

```diff
@@ -22,6 +22,28 @@
 #include "base/threading/thread_local.h"
 #include "build/build_config.h"
 
+#include <dlfcn.h>
+
+static void* gRecordReplayAssertFn;
+
+static void RecordReplayAssert(const char* aFormat, ...) {
```

Comment:
Why does this have to be directly defined here like this?

Reply:
There are link errors if we call the recordreplay:: methods directly, this has happened in a few other places in chromium's core code.